src/main/java/org/broadinstitute/hellbender/engine/Shard.java

```diff
@@ -0,0 +1,87 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * An interface representing a group of reads with an associated
+ */
+public interface Shard<T> extends Iterable<T>,Locatable {
+
+    /**
+     * @return the interval this shard spans
+     */
+    SimpleInterval getInterval();
+
+    /**
+     * @return the interval this shard spans, potentially with additional padding on each side
+     * it must be the case that for a given Shard getPaddedInterval().contains(getInterval())
+     */
+    SimpleInterval getPaddedInterval();
+
+    /**
+     * @return the start of the non-padded interval this shard covers
+     */
+    @Override
+    default int getStart(){
+        return getInterval().getStart();
+    }
+
+    /**
+     * @return the end of the non-padded interval this shard covers
+     */
+    @Override
+    default int getEnd(){
+        return getInterval().getEnd();
+    }
+
+    /**
+     * @return contig this shard belongs to
+     */
+    @Override
+    default String getContig(){
+        return getInterval().getContig();
+    }
+
+    /**
+     * Divide an interval into ShardBoundaries. Each shard will cover up to shardSize bases, include shardPadding
+     * bases of extra padding on either side, and begin shardStep bases after the previous shard.
+     *
+     * @param interval interval to shard; must be on the contig according to the provided dictionary
+     * @param shardSize desired shard size; intervals larger than this will be divided into shards of up to this size
+     * @param shardStep each shard will begin this many bases away from the previous shard
+     * @param shardPadding desired shard padding; each shard's interval will be padded on both sides by this number of bases (may be 0)
+     * @param dictionary sequence dictionary for reads
+     * @return List of {@link ShardBoundary} objects spanning the interval
+     */
+    static List<ShardBoundary> divideIntervalIntoShards(final SimpleInterval interval, final int shardSize, final int shardStep, final int shardPadding, final SAMSequenceDictionary dictionary ) {
```

Comment:
Add a trivial overload that does not take `shardStep`, assumes `shardSize == shardStep`, and calls into this version (for symmetry with the version in `LocalReadShard`)